probe-rs/src/architecture/arm/core/armv7a.rs

```diff
@@ -1129,20 +1276,24 @@ impl MemoryInterface for Armv7a<'_> {
     }
 
     fn read_8(&mut self, address: u64, data: &mut [u8]) -> Result<(), Error> {
-        self.halted_access(|core| {
-            for (i, byte) in data.iter_mut().enumerate() {
-                *byte = core.read_word_8(address + (i as u64))?;
-            }
-
-            Ok(())
-        })
+        self.read(address, data)
     }
 
     fn read(&mut self, address: u64, data: &mut [u8]) -> Result<(), Error> {
         self.halted_access(|core| {
             if address % 4 == 0 && data.len() % 4 == 0 {
                 // Avoid heap allocation and copy if we don't need it.
-                core.read_mem_32bit(address, data)?;
+                if let Ok((aligned_buffer, _)) =
+                    <[u32]>::mut_from_prefix_with_elems(data, data.len() / 4)
```

Comment:
When can this fail?

Reply:
Ah, when data is not 4-byte aligned, of course.

Reply:
Yeah, it happened sometimes in my testing. It might be worth it to flag it as an error and fix things that do that, but it happens rarely enough that it didn't seem worth fixing.